File: tests/bench_heavy.py
import time

def sum_to(n):
    s = 0
    for i in range(n + 1):
        s += i
    return s

def is_prime(n):
    if n < 2:
        return 0
    i = 2
    while i * i <= n:
        if n % i == 0:
            return 0
        i += 1
    return 1

def count_primes(limit):
    count = 0
    for i in range(2, limit + 1):
        if is_prime(i):
            count += 1
    return count

def fib(n):
    if n <= 1:
        return n
    return fib(n - 1) + fib(n - 2)

start = time.time()
r1 = sum_to(10000000)
r2 = count_primes(50000)
r3 = fib(35)
end = time.time()
print(r1)
print(r2)
print(r3)
print(f"Time: {(end - start) * 1000:.1f} ms")

File: tests/bench_heavy.cpp
#include <iostream>
#include <chrono>

int sum_to(int n) {
    int s = 0;
    for (int i = 0; i <= n; i++) s += i;
    return s;
}

int is_prime(int n) {
    if (n < 2) return 0;
    for (int i = 2; i * i <= n; i++)
        if (n % i == 0) return 0;
    return 1;
}

int count_primes(int limit) {
    int count = 0;
    for (int i = 2; i <= limit; i++)
        if (is_prime(i)) count++;
    return count;
}

int fib(int n) {
    if (n <= 1) return n;
    return fib(n - 1) + fib(n - 2);
}

int main() {
    auto start = std::chrono::high_resolution_clock::now();
    int r1 = sum_to(10000000);
    int r2 = count_primes(50000);
    int r3 = fib(35);
    auto end = std::chrono::high_resolution_clock::now();
    auto ms = std::chrono::duration_cast<std::chrono::microseconds>(end - start).count() / 1000.0;
    std::cout << r1 << std::endl << r2 << std::endl << r3 << std::endl;
    std::cout << "Time: " << ms << " ms" << std::endl;
    return 0;
}
